extractuser.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"crypto/rsa"
+
+	"golang.org/x/net/context"
+
+	"net/http"
+
+	"github.com/almighty/almighty-core/token"
+	"github.com/goadesign/goa"
+)
+
+// ConfigureExtractUser is a configuration for middleware,
+// Accepts PublicKey and PrivateKey to create a token manager
+// Using token manager, it is responsible for extracting the token from context if present
+// Update the context with uuid found in token
+func ConfigureExtractUser(publicKey *rsa.PublicKey, privateKey *rsa.PrivateKey) (ExtractUser goa.Middleware) {
+	manager := token.NewManager(publicKey, privateKey)
+	return func(h goa.Handler) goa.Handler {
+		return func(ctx context.Context, rw http.ResponseWriter, req *http.Request) error {
+			// for now ignore the error becasue still test for logged in user is not done.
+			uuid, err := manager.Locate(ctx)
+			var value string
+			if err != nil {
+				value = "guest user"
+			}
+			value = uuid.String()
+			ctxWithUser := context.WithValue(ctx, "uuid", value)
```

Comment:
Maybe extract it out into two functions, one for creating the context with the value and one for getting the value. Then than key can be secret between those two functions.

e.g. 

ctx := login.WithIdentity(ctx, Identity)

uuid := login.ContextIdentity(ctx)